app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_counter(user_id = author_id)
+    user = User.find(user_id)
+    user.increment!(:postscounter)
+    user.save
+  end
+
+  after_save :update_counter
+
+  def recent_comments
+    Comment.where(post_id: id).order(created_at: :desc).limit(5)
+  end
```

Comment:
[OPTIONAL]

- So nice to see that you have used one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_counter` method every time a new post has been saved to the database. 

But now you can make the `update_counter` method private which will increase the security.

```ruby
# app/models/post.rb
  after_save :update_counter

  private

  def update_counter
    user.increment!(:posts_counter)
  end
```